auth.go

```diff
@@ -14,7 +15,7 @@ import (
 // This encapsulates the different authentication schemes in use
 type Authenticator interface {
 	// Request creates an http.Request for the auth - return nil if not needed
-	Request(*Connection) (*http.Request, error)
+	Request(context.Context, *Connection) (*http.Request, error)
 	// Response parses the http.Response
 	Response(resp *http.Response) error
```

Comment:
Yeah, I looked at it but there is no communication to the Swift AFAIK?
Just decoding the JSON response and none of the libraries accepts the context, so I personally do not see the need to pass the context here. Maybe just of consistency in the naming and signatures but not sure if that's worth it?

Or am I missing something in the code path?

Reply:
The context is a useful way of passing in things unknown to us now, so I think it would be a good idea to add it. For example the library could set a context key with configuration which this call could read.

Reply:
Ok, I personally tend to avoid using context for anything else than cancellation propagation, but I get the reasoning.
Should be added now https://github.com/ncw/swift/pull/164/commits/5b2a62383c3ca397088e756480cf9810ab297cc0